chia/consensus/block_body_validation.py

```diff
@@ -142,6 +145,12 @@ async def validate_block_body(
     if set(block.transactions_info.reward_claims_incorporated) != expected_reward_coins:
         return Err.INVALID_REWARD_COINS, None
 
+    if block.foliage_transaction_block.timestamp > constants.INITIAL_FREEZE_END_TIMESTAMP:
+        if len(block.transactions_info.reward_claims_incorporated) != len(expected_reward_coins):
```

Comment:
why is this check only happening after the freeze?
since it's a consensus change?

Reply:
Before the TX freeze we don't know if people are adding duplicates or not, and we dont want to cause a fork